From 55db8af0796abc49fdb4202fc42826c422dd95a8 Mon Sep 17 00:00:00 2001
From: Sushanth Sowmyan <khorgath@gmail.com>
Date: Mon, 2 May 2016 12:09:09 -0700
Subject: [PATCH 0759/1164] CDH-44637: HIVE-13645 : Beeline needs null-guard
 around hiveVars and hiveConfVars read (Sushanth
 Sowmyan, reviewed by Thejas Nair)

Change-Id: If67a86af56f6b73084dbb26cd5f3acd918a67b39
---
 .../apache/hive/beeline/DatabaseConnection.java    |   12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/beeline/src/java/org/apache/hive/beeline/DatabaseConnection.java b/beeline/src/java/org/apache/hive/beeline/DatabaseConnection.java
index 02b2eec..ed4318e 100644
--- a/beeline/src/java/org/apache/hive/beeline/DatabaseConnection.java
+++ b/beeline/src/java/org/apache/hive/beeline/DatabaseConnection.java
@@ -128,13 +128,17 @@ boolean connect() throws SQLException {
     }
 
     Map<String, String> hiveVars = beeLine.getOpts().getHiveVariables();
-    for (Map.Entry<String, String> var : hiveVars.entrySet()) {
-      info.put(HIVE_VAR_PREFIX + var.getKey(), var.getValue());
+    if (hiveVars != null){
+      for (Map.Entry<String, String> var : hiveVars.entrySet()) {
+        info.put(HIVE_VAR_PREFIX + var.getKey(), var.getValue());
+      }
     }
 
     Map<String, String> hiveConfVars = beeLine.getOpts().getHiveConfVariables();
-    for (Map.Entry<String, String> var : hiveConfVars.entrySet()) {
-      info.put(HIVE_CONF_PREFIX + var.getKey(), var.getValue());
+    if (hiveConfVars != null){
+      for (Map.Entry<String, String> var : hiveConfVars.entrySet()) {
+        info.put(HIVE_CONF_PREFIX + var.getKey(), var.getValue());
+      }
     }
 
     if (isDriverRegistered) {
-- 
1.7.9.5

